joomla/jissues
# Refactor JoomlacmsPullsListener::checkFilesAndAssignCategory() method

#### Summary of Changes

* Refactor JoomlacmsPullsListener::checkFilesAndAssignCategory()
* Add a unit test for this method

#### Testing Instructions
Check if all files are correct for the corresponding category.
I believe that there are some inconsistencies.
Have a look at the unit test and see if the categories corresponding to the file/path are correct.

**Note** Unit test are failing if executed from within the jtracker script. If they run "standalone" they produce no errors.

@mbabker I'm almost sure that you have a solution to the error:
`ERROR: Serialization of 'Closure' is not allowed`
that appears here: https://travis-ci.org/joomla/jissues/jobs/146807076
Note that unit test run again here with no errors...
